src/react/src/components/DialogTooltip.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import { shape, string, node } from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Tooltip from '@material-ui/core/Tooltip';
+import COLOURS from '../util/COLOURS';
+
+function arrowGenerator(color) {
+    return {
+        '&[x-placement*="bottom"] $arrow': Object.freeze({
+            top: 0,
+            left: 0,
+            marginTop: '-0.95em',
+            width: '3em',
+            height: '1em',
+            '&::before': Object.freeze({
+                borderWidth: '0 1em 1em 1em',
+                borderColor: `transparent transparent ${color} transparent`,
+            }),
+        }),
+        '&[x-placement*="top"] $arrow': Object.freeze({
+            bottom: 0,
+            left: 0,
+            marginBottom: '-0.95em',
+            width: '3em',
+            height: '1em',
+            '&::before': Object.freeze({
+                borderWidth: '1em 1em 0 1em',
+                borderColor: `${color} transparent transparent transparent`,
+            }),
+        }),
+        '&[x-placement*="right"] $arrow': Object.freeze({
+            left: 0,
+            marginLeft: '-0.95em',
+            height: '3em',
+            width: '1em',
+            '&::before': Object.freeze({
+                borderWidth: '1em 1em 1em 0',
+                borderColor: `transparent ${color} transparent transparent`,
+            }),
+        }),
+        '&[x-placement*="left"] $arrow': Object.freeze({
+            right: 0,
+            marginRight: '-0.95em',
+            height: '3em',
+            width: '1em',
+            '&::before': Object.freeze({
+                borderWidth: '1em 0 1em 1em',
+                borderColor: `transparent transparent transparent ${color}`,
+            }),
+        }),
+    };
+}
+
+const styles = () =>
+    Object.freeze({
+        arrow: Object.freeze({
+            position: 'absolute',
+            fontSize: 6,
+            width: '3em',
+            height: '3em',
+            '&::before': Object.freeze({
+                content: '""',
+                margin: 'auto',
+                display: 'block',
+                width: 0,
+                height: 0,
+                borderStyle: 'solid',
+            }),
+        }),
+        bootstrapPopper: Object.freeze(arrowGenerator(COLOURS.DARK_SLATE_GREY)),
+        bootstrapTooltip: Object.freeze({
+            fontSize: '14px',
+            backgroundColor: COLOURS.DARK_SLATE_GREY,
+        }),
+        bootstrapPlacementLeft: Object.freeze({
+            margin: '0 8px',
+        }),
+        bootstrapPlacementRight: Object.freeze({
+            margin: '0 8px',
+        }),
+        bootstrapPlacementTop: Object.freeze({
+            margin: '8px 0',
+        }),
+        bootstrapPlacementBottom: Object.freeze({
+            margin: '8px 0',
+        }),
```

Comment:
Can you rename these styles without using `bootstrap`?
There exist some similar styles `makePreviousOsIdTooltipStyles`. Please check if they can be reused.

Reply:
Done, pls check.